src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java

```diff
@@ -1747,17 +1737,29 @@ public void testInputAnnotationScopeIndentationCheck() throws Exception {
         verifyWarns(checkConfig, fileName, expected);
     }
 
-    @Override
-    protected String internalGetCheckMessage(
-            String messageBundle, String messageKey, Object... arguments) {
-        final ResourceBundle resourceBundle = ResourceBundle.getBundle(
-                messageBundle,
-                Locale.ROOT,
-                Thread.currentThread().getContextClassLoader(),
-                new LocalizedMessage.Utf8Control());
-        final String pattern = resourceBundle.getString(messageKey);
-        final MessageFormat formatter = new MessageFormat(pattern, Locale.ROOT);
-        return formatter.format(arguments);
+    @Test
+    public void testPropertyOrderOfErrorMessages() {
+        final String[] properties = {"##0##", "##1##", "##2##"};
+        final String[] messages = {
+            getCheckMessage(MSG_ERROR, (Object[]) properties),
+            getCheckMessage(MSG_CHILD_ERROR, (Object[]) properties),
+            getCheckMessage(MSG_ERROR_MULTI, (Object[]) properties),
+            getCheckMessage(MSG_CHILD_ERROR_MULTI, (Object[]) properties),
+        };
+        final boolean isInOrder = Arrays.stream(messages).allMatch(msg -> {
+            boolean ordered = true;
+            int prev = -1;
+            for (String property : properties) {
+                final int index = msg.indexOf(property);
+                if (prev >= index) {
+                    ordered = false;
+                    break;
+                }
+                prev = index;
+            }
+            return ordered;
+        });
+        assertTrue("the error messages are required to be in property order", isInOrder);
```

Comment:
It is to verify the properties of error messages is in numerical order, since other tests rely on it. I add this UT as @rnveach suggested in https://github.com/checkstyle/checkstyle/issues/4003#issuecomment-286754290:
> Let's make a test that verifies message is in property order. Add assertion that if it fails, message must be changed so properties are in order.